publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java

```diff
@@ -632,9 +634,17 @@ public void send(String routingKey, String msg, String eventId)
                     : MessageProperties.BASIC;
 
         try {
+            LocalDateTime publishStartTime = LocalDateTime.now();
             channel.basicPublish(exchangeName, routingKey, msgProps, msg.getBytes());
-            log.info("Published message {} with size {} bytes on exchange '{}' with routing key '{}'", eventId,
-                    msg.getBytes().length, exchangeName, routingKey);
+            LocalDateTime publishEndTime = LocalDateTime.now();
+            Duration diff = Duration.between(publishStartTime, publishEndTime);
+            String duration = String.format("%d:%02d:%02d.%03d",
+                    diff.toHours(),
+                    diff.toMinutesPart(),
+                    diff.toSecondsPart(),
+                    diff.toMillisPart());
+            log.info("Published message {} with size {} bytes on exchange '{}' with routing key '{}' with the duration of {}", eventId,
+                    msg.getBytes().length, exchangeName, routingKey, duration);
```

Comment:
I don't think LocalDateTime is a great choice when you want to measure the duration between two events. Since it denotes a timezone-less local time it'll behave quite badly during DST shifts, and in general when the system's time changes. Prefer a monotonic clock like System.nanoTime().

I also don't think logging time series data like this is a great idea in general. It would be much more useful to expose a proper numerical metric (e.g. via OpenTelemetry or Prometheus) that interested parties can pick up.

Don't you want to include the time required to get the message confirmed by the broker?

Reply:
Yes, please, use nanoTime(), as described, for example, here: https://www.baeldung.com/java-measure-elapsed-time. I think it's the most effective method to measure method time.

The general idea here is to help monitor liveness/healthness of communication line to MB and/or the MB itself. It was originated by people managing REMReM/MB components.

There were some tests performed with Micrometer. They offer good data, but they have a statistical nature. However, when an issue related to particular event is to be investigated, it publish time may come very handy.

Reply:
> I don't think LocalDateTime is a great choice when you want to measure the duration between two events. Since it denotes a timezone-less local time it'll behave quite badly during DST shifts, and in general when the system's time changes. Prefer a monotonic clock like System.nanoTime().
> 
> I also don't think logging time series data like this is a great idea in general. It would be much more useful to expose a proper numerical metric (e.g. via OpenTelemetry or Prometheus) that interested parties can pick up.
> 
> Don't you want to include the time required to get the message confirmed by the broker?

Done

Reply:
> Yes, please, use nanoTime(), as described, for example, here: https://www.baeldung.com/java-measure-elapsed-time. I think it's the most effective method to measure method time.
> 
> The general idea here is to help monitor liveness/healthness of communication line to MB and/or the MB itself. It was originated by people managing REMReM/MB components.
> 
> There were some tests performed with Micrometer. They offer good data, but they have a statistical nature. However, when an issue related to particular event is to be investigated, it publish time may come very handy.

Done